src/overlays/actors/ovl_Obj_Bean/z_obj_bean.c

```diff
@@ -54,137 +117,881 @@ static ColliderCylinderInit D_80B90E54 = {
+    this->nextPointIndex = 1;
+}
+
+void ObjBean_SetupPath(ObjBean* this, GlobalContext* globalCtx) {
+    Path* path;
+
+    path = &globalCtx->setupPathList[(this->dyna.actor.params >> 8) & 0x1F];
+    Math_Vec3s_ToVec3f(&this->pathPoints, SEGMENTED_TO_VIRTUAL(path->points));
+}
+
+#ifdef NON_MATCHING
+// Regalloc near speed > mag.
+// f12 vs f2 regs
+void ObjBean_FollowPath(ObjBean* this, GlobalContext* globalCtx) { // Fly/follow path?
+    Path* path;
+    Vec3f acell;
+    Vec3f pathPointsFloat;
+    s32 pad;
+    Vec3s* nextPathPoint;
+    Vec3s* currentPoint;
+    Vec3s* sp4C;
+    Vec3f sp40;
+    Vec3f sp34;
+    f32 sp30;
+    f32 mag;
+    f32 speed;
+
+    Math_StepToF(&this->dyna.actor.speedXZ, sBeanSpeeds[this->unk_1F6].velocity, sBeanSpeeds[this->unk_1F6].accel);
+    path = &globalCtx->setupPathList[(this->dyna.actor.params >> 8) & 0x1F];
+    nextPathPoint = &((Vec3s*)SEGMENTED_TO_VIRTUAL(path->points))[this->nextPointIndex];
+
+    Math_Vec3s_ToVec3f(&pathPointsFloat, nextPathPoint);
+
+    Math_Vec3f_Diff(&pathPointsFloat, &this->pathPoints, &acell);
+    mag = Math3D_Vec3fMagnitude(&acell);
+    speed = CLAMP_MIN(this->dyna.actor.speedXZ, 0.5f);
+    if (speed > mag) {
+        currentPoint = &((Vec3s*)SEGMENTED_TO_VIRTUAL(path->points))[this->currentPointIndex];
+
+        Math_Vec3f_Copy(&this->pathPoints, &pathPointsFloat);
+        this->currentPointIndex = this->nextPointIndex;
+
+        if (this->pathCount <= this->currentPointIndex) {
+            this->nextPointIndex = 0;
+        } else {
+            this->nextPointIndex++;
+        }
+        sp4C = &((Vec3s*)SEGMENTED_TO_VIRTUAL(path->points))[this->nextPointIndex];
+        Math_Vec3s_DiffToVec3f(&sp40, nextPathPoint, currentPoint);
+        Math_Vec3s_DiffToVec3f(&sp34, sp4C, nextPathPoint);
+        if (Math3D_CosOut(&sp40, &sp34, &sp30) != 0) {
+            this->dyna.actor.speedXZ = 0.0f;
+        } else {
+            this->dyna.actor.speedXZ *= (sp30 + 1.0f) * 0.5f;
+        }
+    } else {
+        Math_Vec3f_Scale(&acell, speed / mag);
+        this->pathPoints.x += acell.x;
+        this->pathPoints.y += acell.y;
+        this->pathPoints.z += acell.z;
+    }
+}
+#else
+#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Bean/ObjBean_FollowPath.s")
+#endif
+
+s32 ObjBean_CheckForHorseTrample(ObjBean* this, GlobalContext* globalCtx) {
+    Actor* it;
+
+    it = globalCtx->actorCtx.actorLists[ACTORCAT_BG].head;
+
+    while (it != NULL) {
+        if ((it->id == ACTOR_EN_HORSE) &&
+            (Math3D_Vec3fDistSq(&it->world.pos, &this->dyna.actor.world.pos) < 10000.0f)) {
+            return true;
+        }
+        it = it->next;
+    }
+
+    return false;
+}
+
+void ObjBean_Break(ObjBean* this, GlobalContext* globalCtx) {
+    Vec3f pos;
+    Vec3f velocity;
+    f32 temp_f20;
+    s16 temp_s1;
+    s32 scale;
+    s32 i;
+    s16 gravity;
+    s16 arg5;
+
+    temp_s1 = 0;
+    for (i = 0; i < 36; i++) {
+        temp_s1 += 0x4E20;
```

Comment:
Why decimal? Its an angle.

Reply:
we dont enforce hex or decimal for angles, it can make sense with either or. this is a pretty even number as decimal
also relevant here https://github.com/zeldaret/oot/pull/823#discussion_r640025210